### PR TITLE
Sync dev docs

### PR DESCRIPTION
**Pull request by a-hariti**
Port the latest commits to the develop docs repo

Cutoff point:

```
Author: joshuarli <joshuarli98@gmail.com>
Date:   Wed Aug 14 10:46:08 2024 -0700

    self-hosted: update upgrade path (#1345)
```

### CODE REVIEW

**Review by joshuarli — APPROVED** [review]

File: develop-docs/sdk/event-payloads/replay-recording.mdx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+: **string** The tag is used to describe the "type" of custom event. This will determine how the event gets used in the UI.
+
+`payload`
+
+: **object** The `payload` is similar to the upper level `data` attribute, whose schema is dependent on the `tag`.
+
+In addition to the TypeScript references linked above, the following sections will describe the expected `payload` for custom events.
+
+### options
```
Comment (lizokm) [inline develop-docs/sdk/event-payloads/replay-recording.mdx:47]:
```suggestion
### Options
```

File: develop-docs/sdk/event-payloads/replay-recording.mdx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      "networkDetailHasUrls": false,
+      "networkCaptureBodies": true,
+      "networkRequestHasHeaders": true,
+      "networkResponseHasHeaders": true
+    }
+  }
+}
+```
+
+### breadcrumb
```
Comment (lizokm) [inline develop-docs/sdk/event-payloads/replay-recording.mdx:77]:
```suggestion
### Breadcrumbs
```

File: develop-docs/application/feature-flags/flagpole.mdx [hunk 3]
```diff
@@ -176,22 +181,37 @@ features.add(
...
 
+<Alert level="info">
+    The feature config should not be merged until the registration commit in step 1 has been fully deployed to all target environments. This is because Options Automator will fail to push options to any environments missing the option registration.
+    If this happens, make sure to rerun the options deployment once all environments have been updated to ensure your feature is active.
+</Alert>
+
 Once the option change is deployed, the feature checks will immediately be active in all environments and regions.
 
-_Note:_ The feature config should not be merged until the registration commit in step 1 has been fully deployed to all target environments. This is because Options Automator will fail to push options to any environments missing the option registration.
+## Using flagpole in single-tenants
```
Comment (lizokm) [inline develop-docs/application/feature-flags/flagpole.mdx:191]:
```suggestion
## Using Flagpole in single-tenants
```

**Review by lizokm — APPROVED** [review]
LGTM

**Comment by lforst** [comment]
@a-hariti we need this out rather urgently. I noticed that https://github.com/getsentry/develop/pull/1332 is not included in the docs yet which is the first commit that's not in this PR. Can you double check that everything is moved over?

**Comment by joshuarli** [comment]
Added these.

develop@bf34a3b60cd182f0a59ea2e62c2b22dc120de12d
develop@5abf76ae7da49adb440f9768f1bce12bc223efa2

But, there's actually more than that that's missing....

**Pull request merged**